internal/server/server.go

```diff
@@ -4,19 +4,42 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/felixge/httpsnoop"
+	"github.com/gorilla/mux"
+	"github.com/kanopy-platform/cdnvalidator/internal/server/api/v1beta1"
+	"github.com/kanopy-platform/cdnvalidator/internal/server/middleware/authorization"
 )
 
 type Server struct {
-	router *http.ServeMux
+	router         *mux.Router
+	authCookieName string
 }
 
-func New() http.Handler {
-	s := &Server{router: http.NewServeMux()}
+func New(opts ...Option) (http.Handler, error) {
+	s := &Server{router: mux.NewRouter()}
+
+	for _, opt := range opts {
+		if err := opt(s); err != nil {
+			return nil, err
+		}
+	}
+
+	s.router.Use(logRequestHandler)
 
 	s.router.HandleFunc("/", s.handleRoot())
 	s.router.HandleFunc("/healthz", s.handleHealthz())
 
-	return s.router
+	authmiddleware := authorization.New(authorization.WithCookieName(s.authCookieName),
+		authorization.WithAuthorizationHeader())
+
+	api := v1beta1.New(s.router.PathPrefix(v1beta1.PathPrefix).Subrouter())
```

Comment:
Seems like this `PathPrefix` call should just be the default in the api constructor.

Reply:
Yeah, makes sense.